### DISCUSSION TITLE
Need help with tailwind styles not being applied

### DISCUSSION THREAD

**Discussion by hyphen81** [filamentphp/filament#9249] [Help]
Probably a stupid question, but I'm new to tailwind and filament both. I created a custom theme in using the script provided in the docs, which puts a `tailwind.config.js` and `theme.css` files in the `resources/css/filament/app` directory. 

I've run `npm run dev` and also tried `npm run build`  to compile the assets with vite, but my view isn't seeing the tailwindcss styles, or at least not all of them.

In my blade view I'm using:
```
<link href="{{ asset('css/filament/filament/app.css') }}" rel="stylesheet" />
```
but if I sub that out and reference the CDN, everything works as expected:

```
<script src="https://cdn.tailwindcss.com?plugins=typography" />
```

**Comment by VincvanAs** [comment]
Hi, 

Could you share some more details on your current structure? Like the vite.config.js and postcss.config.js from your application root

Also did you follow the following steps correctly:

```
⇂ First, add a new item to the `input` array of `vite.config.js`: `resources/css/filament/app/theme.css`
⇂ Next, register the theme in the admin panel provider using `->viteTheme('resources/css/filament/app/theme.css')`
⇂ Finally, run `npm run build` to compile the theme
```

**Comment by hyphen81** [comment]
Yes, I did all those steps, and the theme works for my filament app. This page is a view that I'm ultimately using to generate a PDF. It's not within the filament panel or anything like that, it's just a route in my app that I'm using to test this view. I just thought I could reference the css file in the same way I would reference any regular file, but there's some piece I'm not understanding.

**Reply by VincvanAs** [reply]
The asset function in Laravel is used to generate URLs for files in the public directory. If your app.css file is in the resources/css folder, it won't be accessible from the web until you compile it into the public directory

**Reply by hyphen81** [reply]
I did build, and I see the build folder in my public folder. When I open dev tools and look at the file referenced in the html, it's the right location, and _some_ of the styles appear to be there, just not all of them. So I feel like it's not building right or it's not picking up the files in my config.

In my tailwindw.config.js file I have:

```
import preset from '../../../../vendor/filament/filament/tailwind.config.preset'

export default {
    presets: [preset],
    content: [
        './app/Filament/Admin/**/*.php',
        './resources/views/pdf/*.blade.php', <---- this is where my view lives that isn't getting picked up.
        './resources/views/filament/admin/**/*.blade.php',
        './resources/views/filament/app/resources/**/**/*.blade.php',
        './vendor/filament/**/*.blade.php',
    ],
}
```
But the file is definitely in that directory:
![image](https://github.com/filamentphp/filament/assets/10478061/e3aff0ae-eb8f-49a5-8b26-1a1ce702f7c2)

**Reply by sinnbeck** [reply]
Can you show the css classes you are adding that it isnt picking up ? And the code that should be adding them

**Reply by bilogic** [reply]
Is there a way to show which classes were picked up from which file?

**Reply by bilogic** [reply]
While it makes sense that `tailwind.config.js` has to be in the folder of `theme.css` and not in the `base_path`, why isn't `base_path/tailwind.config.js` placed beside `resources/app.css`...... one of the quirks i will never understand

**Reply by shimaxu** [reply]
try this in your pdf template
```@vite('resources/css/filament/app/theme.css')```

**Comment by micraux** [comment]
Same problem for me here, but the PDF doesn't even get generated when I try to use:
`<link href="{{ asset('css/filament/filament/app.css') }}" rel="stylesheet" />`

The PDF does render OK if I use just inline CSS or CSS inside `<style>` in `<head>`.

Any tip/advice/clue?

**Comment by lordjack** [comment]
It worked for me. You need to install Theme Custom as it shows the  [link](https://filamentphp.com/docs/3.x/panels/themes#creating-a-custom-theme), and then you call CSS on component as it shows the example below

You need to make sure tailwind is configured properly (included the filament styles, included app.css etc).

```
@vite('resources/css/app.css') 
<x-filament-panels::page> 
    <div class="grid grid-cols-3 gap-3"> 
        <div class="col-span-2 bg-white rounded border p-5">sd</div> 
        <div class="col-span-1 bg-white rounded border p-5">sd</div> 
    </div> 
</x-filament-panels::page> 
```

**Reply by sourcecodeit** [reply]
Adding @vite('resources/css/app.css') works for me (I didn't make a custom theme), however it's very ugly to manually import the CSS for each component, sounds crazy there's not a better way (I'm sure there is one)

**Reply by humamafif2002** [reply]
its work but make ur form cant be submitted
